plugins/SSH.koplugin/main.lua

```diff
@@ -0,0 +1,154 @@
+local Device =  require("device")
+local InfoMessage = require("ui/widget/infomessage")  -- luacheck:ignore
+local InputDialog = require("ui/widget/inputdialog")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local util = require("util")
+local _ = require("gettext")
+
+--This plugin use a patched dropbear that add two things :
+--  the -n option to allow login without password
+--  read the keyfile from settings/SSH/authorized_keys
+
+if not util.pathExists("dropbearmulti") then return { disabled = true, } end
+
+local SSH = WidgetContainer:new{
+    name = 'SSH',
+    is_doc_only = false,
+}
+
+function SSH:init()
+    self.SSH_port = G_reader_settings:readSetting("SSH_port") or "2222"
+    self.allow_no_password = false
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function SSH:start()
+    local cmd = string.format("%s %s %s %s %s %s",
```

Comment:
Like I said elsewhere in our LuaJIT 2.1 concat (`..`) is also compiled so it doesn't really matter for performance reasons. I reckon concat is easier to edit here with smaller diffs.

Reply:
Perhaps I only said it as an edit? It was a self-correction when I said to use string.format because concat is worse for performance in LuaJIT, but when I linked to http://wiki.luajit.org/NYI#bytecode I saw that because of the upgrade to 2.1 I was saying something outdated and wrong.